### DISCUSSION TITLE
Connect Homespan device to home security video system.

### DISCUSSION THREAD

**Discussion by 350d** [HomeSpan/HomeSpan#410] [Q&A]
Hello! I want to connect ESP32 board to my building video/door security system. I have video terminal inside my apt with buttons to open main entrance and start/stop video call, I can see visitors if they call my apt number etc.
Im looking forward to find and use pins on this video terminal or maybe just a screen backlight power pin as a `DoorBell` device to get notifications and door open button pins as a `Door` device to open the door. Im not sure how to create two devices on single board.

Is that make any sense? Can you guys point me in a right direction how I can do it in a right way? Maybe some project exists already for my specific needs?

**Comment by HomeSpan** [comment]
The simple Doorbell Service is unfortunately no longer supported by Apple - it has been replaced by a more complex protocol that is not supposed by HomeSpan.

I recommend implemented a Contact Sensor service and then using Notifications in the Home App to trigger a notification on your iPhone/MAC whenever the sensor it "tripped".  You test all of this out with a ESP32 and a simple pushbutton switch.  Once you have everything working you can then explore how to interface your video terminal with the ESP32 -- I'm afraid I can't help there.

Note you can implement more than one Accessory on a single device (most of the Example Tutorials contain multiple Accessories) so you can have both a Contact Sensor (to monitor for a doorbell press) as well as a Switch Service that can trigger the door to open.

**Comment by 350d** [comment]
Thank you for this reply. Will try your option as soon as my ESP32 boards will be delivered.
